lmdeploy/vl/model/cogvlm.py

```diff
@@ -61,22 +67,84 @@ def build_model(self):
 
         with disable_logging():
             load_checkpoint_and_dispatch(
-                model=model,
+                model=self.model,
                 checkpoint=self.model_path,
                 device_map=device_map if not self.with_llm else {'': 'cpu'},
                 no_split_module_classes=no_split_module_classes,
                 dtype=torch.half)
-        self.model = model.model.vision
+        self.model = self.model.model.vision
         self.model.eval()
 
+    def preprocess(self, messages: List[Dict]) -> List[Dict]:
+        """refer to the spec of `super().preprocess`"""
+        images = super().collect_images(messages)
```

Comment:
It seems that only father class `VisionModel` has the method `collect_images`, which would be inherited by all its children classes. So it's safe to just use without `super()` to care about the MRO issue
```
images = selfcollect_images(messages)
```